pkg/image-manager/image_builder.go

```diff
@@ -145,92 +129,41 @@ func (b *BaseImageBuilder) dockerStatus() ([]*BaseImage, error) {
 			imageMap[t] = true
 		}
 	}
+	var entities []entitystore.Entity
 	var all []*BaseImage
 	err = b.es.List(b.orgID, nil, &all)
 	if err != nil {
 		return nil, errors.Wrap(err, "Error listing docker images")
 	}
-	for i, bi := range all {
+	for _, bi := range all {
 		url := bi.DockerURL
 		parts := strings.SplitN(url, ":", 2)
 		if len(parts) == 1 {
 			url = fmt.Sprintf("%s:latest", url)
 		}
 
-		status := StatusREADY
 		if _, ok := imageMap[url]; !ok {
-			status = StatusERROR
-		}
-		if bi.Status != status {
-			bi.Status = status
-			rev, err := b.es.Update(bi.Revision, bi)
-			if err != nil {
-				log.Printf("Error updating %s/%s, continue", bi.OrganizationID, bi.Name)
+			// If we are READY, but the image is missing from the
+			// repo, move to ERROR state
+			switch s := bi.Status; s {
+			case entitystore.StatusREADY:
+				bi.Status = entitystore.StatusMISSING
+				entities = append(entities, bi)
 			}
-			bi.Revision = uint64(rev)
-			all[i] = bi
-		}
-	}
-	return all, err
-}
-
-func (b *BaseImageBuilder) poll() error {
-	defer trace.Trace("poll")()
-	baseImages, err := b.dockerStatus()
-	if err != nil {
-		return err
-	}
-	for _, bi := range baseImages {
-		log.Printf("Polling base image %s/%s, delete: %v", bi.OrganizationID, bi.Name, bi.Delete)
-		if bi.Delete {
-			err = b.dockerDelete(bi)
-		}
-		if bi.Status == StatusERROR {
-			err = b.dockerPull(bi)
-		}
-		if err != nil {
-			log.Print(err)
-		}
-	}
-	return nil
-}
-
-func (b *BaseImageBuilder) watch() error {
-	defer trace.Trace("watch")()
-	for {
-		var err error
-		select {
-		case bi := <-b.baseImageChannel:
-			log.Printf("Received base image update %s/%s, delete: %v", bi.OrganizationID, bi.Name, bi.Delete)
-			if bi.Delete {
-				err = b.dockerDelete(&bi)
-			} else {
-				err = b.dockerPull(&bi)
+		} else {
+			// If the image is present, move to READY if in a
+			// non-DELETING statue
+			switch s := bi.Status; s {
+			case entitystore.StatusINITIALIZED,
+				entitystore.StatusCREATING,
+				entitystore.StatusUPDATING,
+				entitystore.StatusERROR:
+				bi.Status = entitystore.StatusREADY
```

Comment:
ugh, formatting on this one is weird. Is this gofmt output? It's hard to see where do the case options end and its code starts.

Reply:
I know... it's gofmt.  I had it more clear, but it just got moved back :(